src/lib.rs

```diff
@@ -92,9 +92,21 @@ impl Default for Type {
     }
 }
 
-/// Builder for the generator.
-#[derive(Default, Clone, Debug, PartialEq)]
-pub struct Builder<F, V>
+/// This is the error type for the `Builder::build` function.
+#[derive(Debug, PartialEq, Eq, Copy, Clone)]
+pub enum ConfigurationError {
```

Comment:
I also don't really like having *Unspecified errors as they are coding errors and there is nothing to do about them run time.
And *MultiplyDefined errors seem odd too as setting value twice isn't nessacary error and can just restrict the APIs usage.
If the errors were encode type system ala [session types](https://scribbles.pascalhertleif.de/elegant-apis-in-rust.html#session-types) I would feel better about them. But that would feel bit too heavy for this.

Have to think about this.